src/matching/team_recommendations.py

```diff
@@ -13,127 +36,84 @@ def get_team_recommendations(email):  # GET
     Return:
         a list of recommended teams to join
     """
+    skills_weight = 1.2
+    seriousness_weight = 1.1
 
     user = coll("users").find_one({"_id": email})
     if not user:
         return {"message": "Invalid user"}, 403
 
     # basic info about users
     skills = user["skills"]
-    interests = user["interests"]
-    prizes = user["prizes"]
     seriousness = user["seriousness"]
 
-    names = set()
-    matches = []
+    all_open_teams = coll("teams").aggregate([{"$match": {"complete": False}}])
+    all_open_members = coll("users").aggregate([{"$match": {"team_id": {"$all": [all_open_teams]}}}])
+
+    team_map = dict()
+
+    # map of distances
+    for member in all_open_members:
+        team_id = member["team_id"]
+        dis = lv_distance(user, member)
+        if team_id in team_map:
+            team_map[team_id] = [dis]
+        else:
+            team_map[team_id].append(dis)
+
+    # average the distance
+    for list_key in team_map:
+        member_list = team_map[list_key]
+        team_map[list_key] = sum(member_list) / float(len(member_list))
 
     # match for skill
     needed_skills = set()
     frontend_languages = set(["html", "css", "javascript", "php", "typscript"])
     backend_languages = set(["java", "php", "ruby", "python", "c", "c++", "sql", "node.js"])
+
     # judging if the user if frontend or backend, and give backend suggestions if only know frontend, vice versa
     skill_set = set(skills)
-    front_num = len(skill_set.intersection(skills))
-    back_num = len(skill_set.intersection(skills))
+    front_num = len(skill_set.intersection(frontend_languages))
+    back_num = len(skill_set.intersection(backend_languages))
 
-    if front_num > (back_num * len(frontend_languages) / len(backend_languages)):
-        if back_num < 3:
-            needed_skills.update(backend_languages)
-    else:
-        if front_num < 3:
+    front_pers = front_num/(front_num+back_num)
+    back_pers = 1-front_pers
+    if front_pers > back_pers:
+        if front_pers < 0.3:
```

Comment:
This number seems arbitrary - is there a reason we picked this? Might be better to just do a direct greater than/less than comparison